Add docs for deprecated generated (#2220)
sorbet/sorbet: website/docs/error-reference.md

end
```

## 5056

> Generated method signatures have been removed from Sorbet.

Reviewer: ```suggestion
The `generated` annotation has been deprecated from Sorbet.
```

Reviewer: Suggesting this because it took me a while to figure out why Sorbet was removing generated method before I actually think about `.generated`.

> The `generated` annotation has been deprecated from Sorbet.

Author: ```suggestion
The `generated` annotation in method signatures is deprecated.
```
